### DISCUSSION TITLE
[QUESTION]:采集容器日志，ilogtail没有输出

### DISCUSSION THREAD

**Discussion by nsdsb123** [alibaba/ilogtail#1177] [Help] [answered]
查看ilogtail日志输出如下：
ilogtail started. pid: 11
register fun v2 0x6b7a20 0x6b9a30 0x6b9280 0x6b9a60
2023/10/09 08:36:05 DEBUG Now using Go's stdlib log package (via loggers/mappers/stdlib).
load log config /usr/local/ilogtail/plugin_logger.xml 
recover stdout
recover stderr

看不到容器输出

**Comment by linrunqi08** [comment]
具体使用的是什么采集配置呢？是K8s场景还是docker场景？

**Comment by nsdsb123** [comment]
apiVersion: v1
kind: ConfigMap
metadata:
  name: ilogtail-user-cm
  namespace: test
data:
  app_stdout.yaml: |
    enable: true
    inputs:
      - Type: service_docker_stdout
        Stderr: false
        Stdout: true                # only collect stdout
        IncludeK8sLabel:
          app: test-gateway                # choose containers with this label
    flushers:
      - Type: flusher_stdout
        OnlyStdout: true


config用的这个配置

**Comment by nsdsb123** [comment]
> 具体使用的是什么采集配置呢？是K8s场景还是docker场景？

k8s场景

**Comment by linrunqi08** [comment]
@nsdsb123  可以按照下面的步骤改下看看：
1. 确认一下你要采集的容器是否包含app: test-gateway这个label
2. Stderr设置成true
3. flusher_stdout改成输出到文件中

**Comment by nsdsb123** [comment, marked as answer]
已经解决了，我测试的时候改了一下官方的例子的yaml文件，所以没成功采集，后面直接用官方的例子就成功了，但是不知道具体是哪些改动导致有问题，下面是我一开始错误的yaml文件，还有个问题想请教一下，用回官方的模板可以采集，但是不太清楚他这个采集的逻辑，我采集这个label是有两个pod的，分别在不同的节点上，然后我得ilogtail部署在6个节点上，但是其中只有3个节点是能正常采集的。

apiVersion: apps/v1
kind: deployment
metadata:
  name: ilogtail-ds
  namespace: backend
  labels:
    k8s-app: logtail-ds
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: logtail-ds
  template:
    metadata:
      labels:
        k8s-app: logtail-ds
    spec:
      containers:
        - name: logtail
          env:
            - name: ALIYUN_LOG_ENV_TAGS       # add log tags from env
              value: _node_name_|_node_ip_
            - name: _node_name_
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: spec.nodeName
            - name: _node_ip_
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: status.hostIP
          image: >-
            sls-opensource-registry.cn-shanghai.cr.aliyuncs.com/ilogtail-community-edition/ilogtail:latest
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - mountPath: /usr/local/ilogtail/checkpoint # for checkpoint between container restart
              name: checkpoint
            - mountPath: /usr/local/ilogtail/user_yaml_config.d # mount config dir
              name: user-config
              readOnly: true
      volumes:
        - hostPath:
            path: /etc/ilogtail-ilogtail-ds/checkpoint
            type: DirectoryOrCreate
          name: checkpoint
        - configMap:
            defaultMode: 420
            name: ilogtail-user-cm
          name: user-config
